website_event_register_free/model/event_registration.py

```diff
@@ -24,15 +24,15 @@
 class EventRegistration(models.Model):
     _inherit = 'event.registration'
 
-    def _prepare_registration(self, event, post, user_id, partner_id=False):
+    def _prepare_registration(self, event, post, user_id, partner=False):
         return {
             'origin': 'Website',
             'nb_register': int(post['tickets']),
             'event_id': event.id,
             'date_open': fields.Datetime.now(),
-            'email': post['email'],
-            'phone': post['phone'],
-            'name': post['name'],
+            'email': partner and partner.email or post['email'],
```

Comment:
You are giving more priority to the store field instead of the possible data filled on the form. You should invert it.
`post.get('email') or partner.email`